PolyEngine/Engine/Src/TransformComponent.cpp

```diff
@@ -24,20 +15,44 @@ void TransformComponent::SetParent(TransformComponent* parent)
 {
 	if (parent != nullptr)
 	{
+		parent->UpdateGlobalTransformationCache();
+		LocalTranslation = LocalTranslation - parent->GlobalTranslation;
+		LocalRotation = LocalRotation * parent->GlobalRotation.GetConjugated();
```

Comment:
I'm not sure about the order of this multiplication, but lets leave it for now. Just keep an eye on it ;)